### PR TITLE
Init

### LINKED ISSUE #1: [2pt] Initial setup |  Back-end
- Use Rails to create backend API.
- You should use Postgres as your database
- The [Creative Commons license of the design](https://creativecommons.org/licenses/by-nc/4.0/) requires that you give appropriate credit to the author. Therefore, you must do it in the README of your project.

### LINKED ISSUE #7: [3pt] Back-end | Models
Create models for:
- Car class:
  - Model
  - City
  - Image
  - Has Reservation
  - Deleted?
 - User:
   - First name
   - Last name
   - Has Reservation
   - Deleted?
- Reservation
  - Belongs to user
  - Belongs to car
  - Reservation date
  - Reservation time
 
ERD for BookTesla

### PR DESCRIPTION
**Pull request by lu-jim**
## Purpose:
- Closes #1 
- Closes #7 

## Features:
- Created basic rails API application
- Configured linters
- Created models for Car, User, and Bookings
- Established their relationships according to the ERD
- Created basic tests for models

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+Rails.application.routes.draw do
```
Comment (Maclenn77) [inline config/routes.rb:1]: Suggestion: Add routes of created models.

File: .stylelintrc.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
+{
```
Comment (Maclenn77) [inline .stylelintrc.json]: Remove this file as it's not needed for a Rails API-only app.

File: config/puma.rb [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+threads min_threads_count, max_threads_count
+
+# Specifies the `worker_timeout` threshold that Puma will use to wait before
+# terminating a worker in development environments.
+#
+worker_timeout 3600 if ENV.fetch("RAILS_ENV", "development") == "development"
+
+# Specifies the `port` that Puma will listen on to receive requests; default is 3000.
+#
+port ENV.fetch("PORT") { 3000 }
```
Comment (Maclenn77) [inline config/puma.rb]: Change port because is the same that origin in cors

**Review by Maclenn77 — APPROVED** [review]
Hi, @lu-jim.

I already reviewed your code. My only comments for now are: 

- Remove .stylelintrc.json as it's not needed for a Rails Api-only app.

I checked that there's not still any routes specified for API. Would it not be part of the setup?

I checked that you already added `gem rake-cors` and edited the cors file. An observation: the origin in the cors file is setup to localhost:3000 and port is also 3000 in Puma. I think it should be a different one in order to run the front and backend separately. Am I right?

Greetings, 
JP

File: config/routes.rb (hunk 1 above)
Comment (lu-jim) [reply config/routes.rb:1]: I think routes make more sense when we have controllers in place. At the moment, they wouldn't do anything as the views are managed from the front-end and any route in place wouldn't return anything.

File: .stylelintrc.json (hunk 2 above)
Comment (lu-jim) [reply .stylelintrc.json]: Done!

File: config/puma.rb (hunk 3 above)
Comment (lu-jim) [reply config/puma.rb]: Thanks for the feedback! I didn't know how to change the default port other than specifying when running `rails s`.

**Pull request merged**